packages/trpc/server/document-router/router.ts

```diff
@@ -172,4 +173,66 @@ export const documentRouter = router({
         });
       }
     }),
+
+  duplicateDocument: authenticatedProcedure
+    .input(ZGetDocumentByIdQuerySchema)
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const { id } = input;
+        const { user } = ctx;
+
+        const document = await prisma.document.findUniqueOrThrow({
+          where: {
+            id,
+            userId: user.id,
+          },
+          select: {
+            title: true,
+            userId: true,
+            documentData: {
+              select: {
+                data: true,
+                initialData: true,
+                type: true,
+              },
+            },
+            documentMeta: {
+              select: {
+                message: true,
+                subject: true,
+              },
+            },
+          },
+        });
+
+        const createdDocument = await prisma.document.create({
+          data: {
+            title: document.title,
+            User: {
+              connect: {
+                id: document.userId,
+              },
+            },
+            documentData: {
+              create: {
+                ...document.documentData,
```

Comment:
You'll want to do the following otherwise we would pull in their sealed changes if the document was complete:

```typescript
{
  ...document.documentData,
  data: document.dataData.initialData,
}
```